WooCommerce/WooCommerceTests/POS/Models/PointOfSaleAggregateModelTests.swift

```diff
@@ -262,4 +264,103 @@ struct PointOfSaleAggregateModelTests {
         // Then
         #expect(sut.itemListState == .error(expectedError))
     }
+
+    struct CartTests {
+        let sut: PointOfSaleAggregateModel
+        private var analytics: WooAnalytics!
+        private var analyticsProvider: MockAnalyticsProvider!
+
+        init() {
+            analyticsProvider = MockAnalyticsProvider()
+            analytics = WooAnalytics(analyticsProvider: analyticsProvider)
+            sut = PointOfSaleAggregateModel(itemProvider: MockPOSItemProvider(),
+                                            analytics: analytics)
+        }
+
+        @Test func addItem_results_in_a_non_empty_cart() async throws {
+            // Given
+            try #require(sut.cart.isEmpty)
+            let item = Self.makeItem()
+
+            // When
+            sut.addToCart(item)
+
+            // Then
+            #expect(sut.cart.isNotEmpty)
+        }
+
+        @Test func addItem_puts_new_items_first_in_the_cart() async throws {
```

Comment:
nit: does this test case need `async throws`? I might be missing some requirements for Swift Testing.

Reply:
No, strictly it doesn't need it, but it's part of the default Test function expansion. 

A lot of tests do want to throw, e.g. when you use the `#require` macro, and similar with async functions being more common. It doesn't really cost us anything to have it there IMO.